fuel-vm/src/interpreter/memory.rs

```diff
@@ -379,11 +625,14 @@ pub(crate) fn push_selected_registers(
 
     // First update the new stack pointer, as that's the only error condition
     let count: u64 = bitmask.count_ones().into();
-    let stack_range = MemoryRange::new(*sp, count * 8)?;
-    try_update_stack_pointer(sp, ssp, hp, stack_range.words().end)?;
+    let write_size = count * (core::mem::size_of::<Word>() as u64);
+    let write_at = *sp;
+    try_update_stack_pointer(sp, ssp, hp, write_at + write_size, memory)?;
```

Comment:
I know that it shouldn't be possible to overflow here, but it would be nice to add `#![deny(clippy::arithmetic_side_effects)]` for `fuel-vm`. But it is definitely not a part of the current PR=)

Reply:
Agreed. It's definitely not too easy to see that this cannot overflow.